$OpenBSD: patch-globalconf_h,v 1.6 2014/09/02 10:52:43 dcoppa Exp $

commit 9846a3274abe2569f342bb68af9b290799a8617e
Author: Uli Schlachter <psychon@znc.in>
Date:   Wed Aug 20 11:40:50 2014 +0200

systray: Only intern the atom once

Let's just save the systray atom and keep it around. Why should we redo this
every time this atom is needed?

Signed-off-by: Uli Schlachter <psychon@znc.in>

commit e21119023473fb45a11beb838d935b1c53cb550d
Author: Uli Schlachter <psychon@znc.in>
Date:   Wed Aug 20 11:47:47 2014 +0200

systray: Only register/unregister when needed

Signed-off-by: Uli Schlachter <psychon@znc.in>

--- globalconf.h.orig	Fri Apr 11 11:07:10 2014
+++ globalconf.h	Tue Sep  2 12:21:42 2014
@@ -121,6 +121,10 @@ typedef struct
     struct
     {
         xcb_window_t window;
+        /** Atom for _NET_SYSTEM_TRAY_%d */
+        xcb_atom_t atom;
+        /** Do we own the systray selection? */
+        bool registered;
         /** Systray window parent */
         drawin_t *parent;
     } systray;
